src/xrCore/LocatorAPI.cpp

```diff
@@ -612,8 +612,7 @@ void CLocatorAPI::ProcessOne(pcstr path, const _finddata_t& entry)
             return;
         if (0 == xr_strcmp(entry.name, ".."))
             return;
-        if (path[xr_strlen(path) - 1] != _DELIMITER || path[xr_strlen(path) - 1] != '/')
```

Comment:
@eagleivg do you remember why you have added this check?

Reply:
Engine filesystem has strong requirements on path correctness.
```
Path can be delimited only by a single delimiter.
Delimiters cannot be mixed. (it's either \ or /)
Folder path must always end with delimiter.

Correct path on Windows: folder\\      (ends by single backslash)
Correct path on Windows: folder\\file  (delimited by single backslash only)

Correct path on Linux:   folder/       (ends by single forward slash)
Correct path on Linux:   folder/file   (delimited by single forward slash only)

Wrong path on Windows:   folder
Wrong path on Windows:   folder/file
Wrong path on Windows:   folder//file
Wrong path on Windows:   folder\\\\file

Wrong path on Linux:     folder
Wrong path on Linux:     folder//file
Wrong path on Linux:	 folder\\file
Wrong path on Linux:	 folder\\\\file
etc...
```

Reply:
So always adding delimiter for directory looks good.
I remember when we replaced logical operator with `&&`, engine fails to load some files, so looks like condition is really expected to be `true`.

Reply:
Not exactly. It looks like this condition is always true on Windows, but Linux filesystem (not the engine one! I mean OS internals) can add a delimiter before we enter this function. Removing this check will result in a double delimiter.

Reply:
Hm, for me it looks independent of OS:
```
if (path[xr_strlen(path) - 1] != _DELIMITER || path[xr_strlen(path) - 1] != '/')
```
to simplify reasoning, replace `path[xr_strlen(path) - 1]` with `A`:
```
if (A != _DELIMITER || A != '/')
```
replace `_DELIMITER` with `2` and '/' with `3` (`_DELIMITER` now is '\\'):
```
if (A != 2 || A != 3)
```
so this condition is always `true`, independent of `A` value.

Do you mean `_DELIMITER` is generated and OS-dependent (`/` for Unix-like or Mac)?
If so, may be replace condition with:
```
#if defined(_WIN32)
   xr_strcat(N, DELIMITER);
#else
if (A != _DELIMITER)
  ...
#endif

?

Reply:
> Not exactly. It looks like this condition is always true on Windows, but Linux filesystem (not the engine one! I mean OS internals) can add a delimiter before we enter this function. Removing this check will result in a double delimiter.

Removed this commit :)